### PR TITLE
Reform documentation for all assignId methods in the Annotator.

### LINKED ISSUE #881: Tidy annotator assignId documentation.
The annotator assignId documentation for the public API contains the following:

```
     * @see assignId(const AnyCellmlElementPtr &item) for a list of reasons that an identifier may not be assigned.
     * @see errorCount() if and/or error(size_t) for any issues that may have been raised.
```

which can be removed as the `@overload` directive covers this.

### PR DESCRIPTION
**Pull request by hsorby**
Fixes #881.

### CODE REVIEW

File: src/api/libcellml/annotator.h [hunk 1]
```diff
@@ -840,25 +829,22 @@ class LIBCELLML_EXPORT Annotator: public Logger
...
-     * @see assignId(const AnyCellmlElementPtr &item) for a list of reasons that an identifier may not be assigned.
-     * @see errorCount() if and/or error(size_t) for any issues that may have been raised.
+     * @see assignId(const AnyCellmlElementPtr &item)
      *
-     * @overload
-     *
-     * @param variable1 A @ref VariablePtr item defining to the first item in a connection or equivalence.
-     * @param variable2 A @ref VariablePtr item defining to the second item in a connection or equivalence.
+     * @param variable1 A @ref VariablePtr item defining to the first item in a equivalence.
+     * @param variable2 A @ref VariablePtr item defining to the second item in a equivalence.
```
Comment (agarny) [inline src/api/libcellml/annotator.h]: "... defining the..." rather than "defining TO the..."

File: src/api/libcellml/annotator.h [hunk 2]
```diff
@@ -675,22 +675,20 @@ class LIBCELLML_EXPORT Annotator: public Logger
...
      * This method will return the new identifier that has been assigned, or an empty string
      * if the operation failed.
      *
      * Only CellmlElementType::MODEL and CellmlElementType::ENCAPSULATION are relevant any other
-     * CellmlElementType type will not assign an identifier.  @see errorCount() if and/or error(size_t) for any
-     * issues that may have been raised.
-     *
-     * @see assignId(const AnyCellmlElementPtr &item) for a list of reasons that an identifier may not be assigned.
-     * @see errorCount() if and/or error(size_t) for any issues that may have been raised.
+     * CellmlElementType type will not assign an identifier.
```
Comment (agarny) [inline src/api/libcellml/annotator.h]: Might be good to have some `@ref` here and there (e.g., for `CellmlElementType` on line 686).

Comment (hsorby) [reply src/api/libcellml/annotator.h]: a7ccc63

File: src/api/libcellml/annotator.h (hunk 1 above)
Comment (hsorby) [reply src/api/libcellml/annotator.h]: a7ccc63

**Review by agarny — APPROVED** [review]

**Review by nickerso — APPROVED** [review]

**Pull request merged**